tests/integration/app-charm/src/helpers.py

```diff
@@ -451,3 +462,83 @@ def check_inserted_data_kyuubi(credentials: Dict[str, str], database_name: str)
         return True
     except Exception:
         return False
+
+
+# ETCD
+
+
+def generate_cert(common_name: str):
+    ca_private_key = generate_private_key()
+    ca_cert = generate_ca(
+        private_key=ca_private_key, validity=timedelta(days=365), common_name="ca_common_name"
+    )
+
+    client_private_key = generate_private_key()
+    client_csr = generate_csr(private_key=client_private_key, common_name=common_name)
+    client_cert = generate_certificate(
+        client_csr, ca_cert, ca_private_key, validity=timedelta(days=365)
+    )
+    return "\n".join([client_cert.raw, ca_cert.raw]), client_private_key.raw
+
+
+def insert_data_etcd(credentials: Dict[str, str], database_name: str) -> bool:
```

Comment:
```suggestion
def insert_data_in_etcd(credentials: Dict[str, str], database_name: str) -> bool:
```
(also in subsequent functions)